core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -490,10 +490,10 @@ void sendBuddyConnectResponseMessage(UUID senderUserId, UUID senderUserAnonymize
 
 	private void disconnectBuddyIfConnected(UserAnonymizedDto buddyUserAnonymized, UUID userAnonymizedId)
 	{
-		Optional<BuddyAnonymized> buddyAnonymized = buddyUserAnonymized.getBuddyAnonymized(userAnonymizedId);
-		buddyAnonymized.ifPresent(ba -> {
-			ba.setDisconnected();
-			BuddyAnonymized.getRepository().save(ba);
+		Optional<BuddyAnonymizedDto> buddyAnonymized = buddyUserAnonymized.getBuddyAnonymized(userAnonymizedId);
+		buddyAnonymized.map(ba -> buddyAnonymizedRepository.findOne(ba.getId())).ifPresent(bae -> {
+			bae.setDisconnected();
+			userAnonymizedService.updateUserAnonymized(buddyUserAnonymized.getId());
```

Comment:
A disadvantage of the caching introduced in this PR is that we have to be very aware that any change to `BuddyAnonymized` has to result in a cache update. Shouldn't this be reflected in a comment on `BuddyAnonymized`?

Reply:
There is nothing wrong with that, but the question is whether it'll be read when the knowledge is required. I'll add a class comment explaining this.